### ISSUE TITLE
pnsr evalaution result on kitti 06 seqence

### ISSUE THREAD

**Issue by LevinJ** [DengKaiCQ/GigaSLAM#7]
Hi I tried the experiment on kitti 06 sequence, with evaluation flag being on. and the resultant psnr value is 17.11, while the expected value in GigaSLAM paper is 24.17.  It may be worthwhile noting that I replaced unitdepth codes by latest version from unidepth repository.

![Image](https://github.com/user-attachments/assets/e620828c-de50-40fa-ae5c-a862ae3abf60)

and below is the trajectory chart generated. The scale does not seem to be right, I assume it should be about 400 meters long.

![Image](https://github.com/user-attachments/assets/20e8645b-e771-4a3d-b574-cee2e9684796)

**Comment by DengKaiCQ** [comment]
I did a bunch of exps today, and i'm pretty sure that the problem along with #6 is caused by the version of `UniDepth`. Here is what i got today:

### 1. Latest version of `UniDepth` (code `8d8cfe4` on GitHub and pre-train weight `52b349b` on Hugging Face):

![Image](https://github.com/user-attachments/assets/64829e96-d96d-4ee9-8cc3-a4aa8b556c35)


<div align="center">
<img src="https://github.com/user-attachments/assets/64c89da2-8209-4510-bc10-f82edef91a89" width="55%">
</div>

You see, I got the same result as you did. So this is probably because the new version of `UniDepth` has something different from the older one, and this matches the result in  [this issue](https://github.com/lpiccinelli-eth/UniDepth/issues/128). The new version of `UniDepth` has a severe scale drift, and most of the depths are within 1.0 m (median depth is 0.75 m). When voxelizing, an abnormal number of points are merged into a single voxel, so the MLPs decode insufficient GS primitives because there aren’t many voxels left.

### 2. Older version of `UniDepth` (code `bebc4b2` on GitHub and pre-train weight `1d0d3c5` on Hugging Face):

If we use the older version of `UniDepth`, the scale of depth is consistent. It is in metric scale, so it works well.

![Image](https://github.com/user-attachments/assets/f7317488-63bf-45a8-80b1-b64e4fe7c892)

<div align="center">
<img src="https://github.com/user-attachments/assets/beaf602d-5e7b-4ed1-b444-e6c4edda4674" width="55%">
</div>

So the correct version of the pretrained weights should be:

**ViT-Large:** `Jun 12, 2024` commit hash `1d0d3c52f60b5164629d279bb9a7546458e6dcc4`
**ViT-Small:** `Jun 12, 2024` commit hash `045c7eff65610869bf4917432ec3ba0d739f0cb7`

I will fix the problem to ensure the code uses the right version of `UniDepth`. The update will be committed in a few hours. However, after hours of debugging, I still have no clue why the new version of `UniDepth` is behaving this way...

**Comment by LevinJ** [comment]
Nice work on the quick identification of root cause!  Yes, I used the old version of unitdepth, and was able to reproduce the pnsr result reported in paper. I did the tests on kitti 06 and 10 sequence.  The kitti odometry score (RPE, evaluated sequence length being [100,200,300,400,500,600,700,800]) is very impressive as well, being 0.8 % and 1.2% separately.

**Comment by DengKaiCQ** [comment]
it's awesome to hear the update works on your machine! Really glad it's running smoothly now. Hope `GigaSLAM` gives your project / research some insights and help! 🙂
